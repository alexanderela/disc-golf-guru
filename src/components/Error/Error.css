.Error {
  align-items: center;
  background-color: #FFF;
  border: 3px solid #febb00;
  border-radius: 20px;
  display: flex;
  flex-direction: column;
  height: 9rem;
  justify-content: center;
  margin: 1.25rem auto;
  padding: 0.75rem 0 1rem 0;
  width: 38rem;
  box-shadow: 0px 0px 1px 2px #Feeb00;
}

.error-header {
  font-size: 1.25rem;
  font-weight: 600;
  color: #2c9a2c;
  margin: 1rem 4rem 0;
}

.error-message {
  margin: 1rem 4rem;
  line-height: 1.25rem;
}
