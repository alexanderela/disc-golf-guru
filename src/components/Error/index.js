import React from 'react';
import './Error.css';

const Error = ({ message }) => (
  <div className="Error">
    <p className="home-search-instructions">Oops!</p>

    <p className="home-intro">{message}</p>
  </div>
);

export default Error;
