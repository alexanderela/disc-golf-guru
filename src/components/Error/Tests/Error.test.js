import React from 'react';
import ReactDOM from 'react-dom';
import Error from '../';
import { shallow } from 'enzyme';

describe('Error', () => {
	let wrapper;

	beforeEach(() => {
		wrapper = shallow(<Error />);
	})

	it('should render like snapshot', () => {
		expect(wrapper).toMatchSnapshot();
	})
})