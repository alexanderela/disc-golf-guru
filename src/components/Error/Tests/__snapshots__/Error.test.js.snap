// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error should render like snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Error />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p
          className="home-search-instructions"
        >
          Oops!
        </p>,
        <p
          className="home-intro"
        />,
      ],
      "className": "Error",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Oops!",
          "className": "home-search-instructions",
        },
        "ref": null,
        "rendered": "Oops!",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "home-intro",
        },
        "ref": null,
        "rendered": null,
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            className="home-search-instructions"
          >
            Oops!
          </p>,
          <p
            className="home-intro"
          />,
        ],
        "className": "Error",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Oops!",
            "className": "home-search-instructions",
          },
          "ref": null,
          "rendered": "Oops!",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "home-intro",
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
